### PR TITLE
Add preset date ranges dropdown filter on payment activity widget.

### PR DESCRIPTION
**Pull request by shendy-a8c**
>[!note]
>The PR was started afresh on #8910 since this was stale and had too many conflicts with `develop`

#### Changes proposed in this Pull Request

Show a dropdown of date range presets on the right of the Payment Activity header.
The dropdown currently does not do anything other than

#### Testing instructions

*

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8730` or branch name `add/8482-preset-date-ranges` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/8482-preset-date-ranges) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 1a199d8eb5efa123f7b5297d92a2d9daf0f37c2c
- Build time: 2024-06-05 08:19:50 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/payment-activity/index.tsx [hunk 1]
```diff
@@ -12,22 +20,220 @@ import interpolateComponents from '@automattic/interpolate-components';
...
+					second: 59,
+					millisecond: 0,
+				} );
+				break;
+			}
+			case 'last_7_days': {
+				const now = moment();
+				start = now
+					.clone()
+					.subtract( 7, 'd' )
```
Comment (shendy-a8c) [inline client/components/payment-activity/index.tsx:138]: Should this be 6 or 7.

Context: p1714639411758769-slack-C06BBFTF6EM

Comment (shendy-a8c) [reply client/components/payment-activity/index.tsx:138]: Should be 7 and date end should be yesterday, not today.

Discussed in https://github.com/Automattic/woocommerce-payments/pull/8751#discussion_r1608051182

**Comment by shendy-a8c** [comment]
Closing this PR since there has been a change to scope pdjTHR-3KJ-p2#comment-5245.
Also, the work continues with @nagpai's  https://github.com/Automattic/woocommerce-payments/pull/8910.

**Pull request closed without merging**